src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -42,8 +43,8 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
         $handle = false;
         $sender = null;
 
-        foreach ($this->sendersLocator->getSenders($envelope, $handle) as $sender) {
-            $envelope = $sender->send($envelope);
+        foreach ($this->sendersLocator->getSenders($envelope, $handle) as $alias => $sender) {
+            $envelope = $sender->send($envelope)->with(new SentStamp(\get_class($sender), \is_string($alias) ? $alias : null));
```

Comment:
the handledstamp receives the handler instance, why not give the sent stamp the sender instance as well?

Reply:
No opinion. I don't see much benefit creating a static factory for this on contrary of the HandledStamp.

Reply:
it would be `public function __construct(SenderInterface $sender` actually

Reply:
looks like binding to one use case, better decouple